pkg/kubelet/kubelet.go

```diff
@@ -2892,6 +2892,24 @@ func (kl *Kubelet) updateRuntimeUp() {
 		return
 	}
 	kl.runtimeState.setRuntimeState(nil)
+
+	// Load optional conditions
+	// https://github.com/kubernetes/cri-api/blob/v0.29.1/pkg/apis/runtime/v1/api.proto#L1496-L1517
+	// > 2. Optional conditions: Conditions are informative to the user, but kubelet
+	// > will not rely on. Since condition type is an arbitrary string, all conditions
+	// > not required are optional. These conditions will be exposed to users to help
+	// > them understand the status of the system.
+	var optionalConditions []kubecontainer.RuntimeCondition
+	for _, f := range s.Conditions {
+		switch f.Type {
+		case kubecontainer.RuntimeReady, kubecontainer.NetworkReady:
+			// NOP
+		default:
+			optionalConditions = append(optionalConditions, f)
+		}
```

Comment:
non-blocking nit: I'd prefer a simple if over the switch assuming that it will result in the same performance for string comparisons.

Reply:
Why is `if` better?
I find `switch` to be more readable. Just my feeling though.

Reply:
I guess it's just a matter of taste, let's leave it as-is. :+1: